src/main/java/org/springframework/samples/petclinic/rest/advice/ExceptionControllerAdvice.java

```diff
@@ -16,94 +16,124 @@
-    @ResponseStatus(code = HttpStatus.NOT_FOUND)
-    @ResponseBody
-    public ResponseEntity<ErrorInfo> handleDataIntegrityViolationException(DataIntegrityViolationException ex) {
-        ErrorInfo errorInfo = new ErrorInfo(ex);
-        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(errorInfo);
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<ProblemDetail> handleMethodArgumentNotValidException(MethodArgumentNotValidException ex, WebRequest request) {
+        String detail = ex.getBindingResult().getAllErrors().stream()
+            .map(this::formatErrorMessage)
+            .collect(Collectors.joining("; "));
+
+        return createProblemDetail(
+            "Validation Error",
+            BAD_REQUEST.value(),
+            detail,
+            request,
+            now()
+        );
     }
 
     /**
-     * Handles exception thrown by Bean Validation on controller methods parameters
+     * Formats the error message from an ObjectError.
      *
-     * @param ex      The thrown exception
-     *
-     * @return an empty response entity
+     * @param error the ObjectError to format
+     * @return a formatted error message
      */
-    @ExceptionHandler(MethodArgumentNotValidException.class)
-    @ResponseStatus(BAD_REQUEST)
-    @ResponseBody
-    public ResponseEntity<ErrorInfo> handleMethodArgumentNotValidException(MethodArgumentNotValidException ex) {
-        BindingErrorsResponse errors = new BindingErrorsResponse();
-        BindingResult bindingResult = ex.getBindingResult();
-        if (bindingResult.hasErrors()) {
-            errors.addAllErrors(bindingResult);
-            return ResponseEntity.badRequest().body(new ErrorInfo("MethodArgumentNotValidException", "Validation failed"));
-        }
-        return ResponseEntity.badRequest().build();
+    private String formatErrorMessage(ObjectError error) {
+        return (error instanceof FieldError fieldError)
+            ? String.format("Field '%s' %s", fieldError.getField(), fieldError.getDefaultMessage())
+            : error.getDefaultMessage();
     }
 
+    /**
+     * Creates a ProblemDetail object with the provided details.
+     *
+     * @param title     the title of the problem
+     * @param status    the HTTP status code
+     * @param detail    the detail message of the problem
+     * @param request   the current WebRequest
+     * @param timestamp the timestamp of the problem occurrence
+     * @return a ResponseEntity with the ProblemDetail and the specified status
+     */
+    private ResponseEntity<ProblemDetail> createProblemDetail(String title, int status, String detail, WebRequest request, OffsetDateTime timestamp) {
+        ProblemDetail problemDetail = ProblemDetail.forStatus(status);
+        problemDetail.setTitle(title);
+        problemDetail.setDetail(detail);
+        problemDetail.setInstance(URI.create(request.getDescription(false).substring(4)));
```

Comment:
**thought**: The "instance" member is a JSON string containing a URI reference that identifies the specific occurrence of the problem. See https://datatracker.ietf.org/doc/html/rfc9457#name-instance

We could host our own problem registry. For the beginning, I propose to use the Smartbear registry: https://problems-registry.smartbear.com/
I propose to postpone this change to another PR.